File: PakingLotDemo/src/strategy/payment/PaymentStrategy.java
package strategy.payment;

import model.ParkingTicket;

/**
 * Strategy for processing payment when a vehicle exits the parking lot.
 */
public interface PaymentStrategy {

    /**
     * Attempts to process the payment for the given fee and ticket.
     *
     * @param fee   the amount to charge
     * @param ticket the parking ticket (for reference/logging)
     * @return true if payment succeeded, false if payment failed
     */
    boolean pay(double fee, ParkingTicket ticket);
}

File: problems/python/ParkingLotSystem/src/strategy/payment/payment_strategy.py
"""Strategy for processing payment when a vehicle exits."""
from __future__ import annotations
from abc import ABC, abstractmethod
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from model.parking_ticket import ParkingTicket


class PaymentStrategy(ABC):
    @abstractmethod
    def pay(self, fee: float, ticket: "ParkingTicket") -> bool:
        """Process payment. Returns True if succeeded, False otherwise."""
        pass
